driver/makedepend.ml

```diff
@@ -17,6 +17,47 @@ open Compenv
 open Parsetree
 module String = Misc.Stdlib.String
 
+module Json = struct
+  let escape_char ppf chr =
+      match chr with
+      | ('\"' | '\\') as c ->
+          Format.fprintf ppf "\\%c" c
+      | '\n' ->
+          Format.fprintf ppf "\\%c" 'n'
+      | '\t' ->
+          Format.fprintf ppf "\\%c" 't'
+      | '\r' ->
+          Format.fprintf ppf "\\%c" 'r'
+      | '\b' ->
+          Format.fprintf ppf "\\%c" 'b'
+      | '\x00' .. '\x1F' | '\x7F' as c ->
+          Format.fprintf ppf "\\u%04X" (Char.code c)
+      | c -> Format.fprintf ppf "%c" c
+
+  let escape_string ppf s = String.iter (escape_char ppf) s
+
+  type t =
+    [
+      | `String of string
+      | `Assoc of (string * t) list
+      | `List of t list
+    ]
+
+  let comma ppf () = Format.fprintf ppf ",@ "
```

Comment:
This could be a local function of `print`, because it is not really useful outside (we wouldn't access `Json.comma` from the outside). You can use

```ocaml
let rec print ppf =
  let comma ... in 
  function
  | `String s -> ...
  | ...
```